bumble/utils.py

```diff
@@ -64,6 +65,48 @@ def deregister(self, emitter):
     return cls
 
 
+_F = TypeVar('_F', bound=Callable)
+
+# -----------------------------------------------------------------------------
+class EventWatcherContext(object):
+    handlers: List[Tuple[EventEmitter, str, Callable[..., Any]]]
+
+    def __init__(self) -> None:
+        self.handlers = []
+
+    def on(
+        self, emitter: EventEmitter, event: str, handler: Optional[_F] = None
+    ) -> Union[_F, Callable[[_F], _F]]:
+        def wrapper(f: _F):
+            self.handlers.append((emitter, event, f))
+            emitter.on(event, f)
+
+        if handler is None:
```

Comment:
suggest: `return wrapper if handler is None else wrapper(handler)` to be more concise.

Reply:
Done